lib/_http_client.js

```diff
@@ -191,6 +191,8 @@ function ClientRequest(options, cb) {
     });
   }
 
+  self._requestOptions = options;
```

Comment:
Tiny nit, but would it be possible to move this up closer to the top of the function. I just scanned through a large block of code to make sure that `options` was always defined.